### PR TITLE
[zh-cn] update '2d collision detection'

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

finish translation

### Related issues and pull requests

part of #14272

### CODE REVIEW

File: files/zh-cn/games/techniques/2d_collision_detection/index.md [hunk 1]
```diff
@@ -1,77 +1,149 @@
...
+title: 2D 碰撞检测
 slug: Games/Techniques/2D_collision_detection
+l10n:
+  sourceCommit: acfe8c9f1f4145f77653a2bc64a9744b001358dc
 ---
 
 {{GamesSidebar}}
 
-检测 2D 游戏中的碰撞算法，依赖于可碰撞物体的形状（例如：矩形与矩形，矩形与圆形，圆形与圆形）。通常情况下，你使用的的简单通用形状，会被称为“hitbox”的实体所覆盖，尽管发生的碰撞并不是像素完美契合的，它看起来也足够好，而且可跨多个实体执行碰撞。本文提供了一系列较为通用的 2D 游戏中碰撞侦测技术。
+检测 2D 游戏中的碰撞算法，依赖于可碰撞物体的形状（例如：矩形与矩形，矩形与圆形，圆形与圆形）。通常情况下，你使用的的简单通用形状，会被称为“hitbox”的实体所覆盖，尽管发生的碰撞并不是像素完美契合的，它看起来也足够好，而且可跨多个实体执行碰撞。本文提供了一系列较为通用的 2D 游戏中碰撞检测技术。
```
Comment (yin1999) [inline files/zh-cn/games/techniques/2d_collision_detection/index.md:10]:
```suggestion
2D 游戏中的碰撞检测算法依赖于可碰撞物体的形状（例如：矩形与矩形、矩形与圆形、圆形与圆形）。通常情况下，你使用的简单通用形状，会被称为“碰撞盒（hitbox）”的实体所覆盖，尽管发生的碰撞并不是像素那样完美契合，但它看起来也足够好，而且可跨多个实体执行碰撞。本文提供了一系列较为通用的 2D 游戏中碰撞检测技术。
```

File: files/zh-cn/games/techniques/2d_collision_detection/index.md [hunk 2]
```diff
@@ -1,77 +1,149 @@
...
+    this.color("green");
+  } else {
+    // 没有碰撞
+    this.color("blue");
+  }
+});
 ```
 
-> **备注：** You can see a [live example of Axis-Aligned Bounding Box collision detection](http://jsfiddle.net/knam8/) on jsFiddle, to illustrate how this code would work in practice. [Here is another example without Canvas or external libraries](https://jsfiddle.net/jlr7245/217jrozd/3/).
+{{ EmbedLiveSample('轴对齐包围盒', '700', '300') }}
```
Comment (yin1999) [inline files/zh-cn/games/techniques/2d_collision_detection/index.md:51]: The live sample has been broken (it seems that the js code is executed before the [lib](https://cdnjs.cloudflare.com/ajax/libs/crafty/0.5.4/crafty-min.js) is loaded. Could file a issue in yari (or trying to fix it)?

File: files/zh-cn/games/techniques/2d_collision_detection/index.md [hunk 3]
```diff
@@ -1,77 +1,149 @@
...
+
+> **备注：** [另一个不使用 Canvas 或外部库的示例](https://jsfiddle.net/jlr7245/217jrozd/3/).
```
Comment (yin1999) [inline files/zh-cn/games/techniques/2d_collision_detection/index.md:53]:
```suggestion
> **备注：** [另一个不使用 Canvas 或外部库的示例](https://jsfiddle.net/jlr7245/217jrozd/3/)。
```

File: files/zh-cn/games/techniques/2d_collision_detection/index.md (hunk 2 above)
Comment (jasonren0403) [reply files/zh-cn/games/techniques/2d_collision_detection/index.md:51]: raised mdn/yari#10273

**Review by yin1999 — APPROVED** [review]
Thank you @jasonren0403

**Pull request merged**